tests/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    sync::LazyLock,
+};
+
+use gclient::{GearApi, WSAddress};
+use gear_core::ids::prelude::*;
+use sails_rs::prelude::*;
+use sp_core::crypto::DEV_PHRASE;
+use tokio::sync::Mutex;
+#[cfg(test)]
+mod historical_proxy;
+#[cfg(test)]
+mod vft_manager;
+#[cfg(test)]
+mod checkpoint_light_client;
+
+type State = (u32, HashMap<&'static [u8], CodeId>);
+
+static LOCK: LazyLock<Mutex<State>> =
+    LazyLock::new(|| Mutex::const_new((1_000, HashMap::new())));
+
+pub const DEFAULT_BALANCE: u128 = 500_000_000_000_000;
+pub struct Connection {
+    pub api: GearApi,
+    pub accounts: Vec<(GearApi, ActorId, [u8; 4], String)>,
+    pub code_ids: Vec<CodeId>,
+    pub gas_limit: u64,
+    pub salt: [u8; 4]
+}
+
+pub async fn connect_to_node(
+    balances: &[u128],
+    program: &str,
+    binaries: &[&'static [u8]],
+) -> Connection {
+    let mut lock = LOCK.lock().await;
+
+    let api = GearApi::dev().await.unwrap();
+    let gas_limit = api.block_gas_limit().unwrap();
+    let code_ids = {
+        let mut res = vec![];
+
+        for &binary in binaries {
+            match lock.1.entry(binary) {
+                Entry::Occupied(entry) => {
+                    println!("code {:p} already uploaded", binary);
+                    res.push(*entry.get());
+                }
+
+                Entry::Vacant(entry) => {
+                    println!("uploading code {:p}", binary);
+                    let code_id = api
+                        .upload_code(binary)
+                        .await
+                        .map(|(code_id, ..)| code_id)
+                        .unwrap_or_else(|_| CodeId::generate(binary));
+                    entry.insert(code_id);
+                    res.push(code_id);
+                }
+            }
+        }
+
+        res
+    };
+    
+    let mut accounts = vec![];
+    
+    let mut osalt = lock.0 ;
+    lock.0 += 1_000;
+    for &balance in balances.iter() {
+        let salt = osalt;
+        osalt += 1;
+        let suri = format!("{DEV_PHRASE}//{program}-{salt}:");
+        let api2 = GearApi::init_with(WSAddress::dev(), suri.clone())
```

Comment:
I suggest to use the following to get an ActorId of the new account:
```
    let pair = Pair::from_string(suri, None).map_err(|e| anyhow!("{e:?}"))?;
    let account = <MultiSignature as Verify>::Signer::from(pair.public()).into_account();
    let account_id: &[u8; 32] = account.as_ref();
```

this should be a slight faster since doesn't create another connection to the node.

Then use `let api = api.with(suri).unwrap();` within the tests